src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1060,6 +1107,34 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
                                 summary='Problem loading the helm existing user supplied values')
         raise CLIInternalError("Problem loading the helm existing user supplied values: " + str(e))
 
+    # Change registry_path to private registry if PCR is set in helm values
+    existing_user_values_flat = utils.flatten(existing_user_values)
+    registry_username = None
+    registry_password = None
+    isContainerRegistryEnabled = existing_user_values_flat.get('global.isCustomRegistryEnabled')
+    if isContainerRegistryEnabled:
+        helm_container_repository = existing_user_values_flat.get('systemDefaultValues.image.repository')
+        registry_path = "{}/azurearck8s/azure-arc-k8sagents:{}".format(helm_container_repository, connected_cluster.agent_version)
```

Comment:
So for pcr scenario, when user does not explicitly pass the agent version, we use the existing pcr version. In that case, we can consider skipping helm upgrade step

Reply:
I believe helm automatically handle this part if the version is same.

Reply:
If agent version is not provided for upgrade in PCR scenario, we can avoid making any unnecessary calls like helm login, helm fetch chart, etc.

Reply:
I have not changed the earlier behavior, Earlier with MCR also we were calling helm upgrade. If we want to do this I think we can create another improvement work item.

Reply:
For, MCR upgrade is valid without agent version, it upgrades to the latest available version.

For PCR, when agent version not specified, the latest available version fetch and upgrade is not supported right?

Reply:
Added the suggested change.